tests/linearizability/watch.go

```diff
@@ -82,40 +82,73 @@ func validateWatchResponses(t *testing.T, responses [][]watchResponse, expectPro
 }
 
 func validateMemberWatchResponses(t *testing.T, responses []watchResponse, expectProgressNotify bool) {
-	var (
-		gotProgressNotify          = false
-		lastEventModRevision int64 = 1 // The event.Kv.ModRevision in the latest event.
-		lastHeadRevision     int64 = 1 // The resp.Header.Revision in last watch response.
-	)
+	// Validate watch is correctly configured to ensure proper testing
+	validateGotProgressNotify(t, responses, expectProgressNotify)
+
+	// Validate etcd watch properties
+	validateDuplicateHeaderRevisionMeansNoEvents(t, responses)
+	validateResponseHeaderRevisionNeverDecreases(t, responses)
+	validateModRevisionIsStrictlyIncremental(t, responses)
+	validateHeaderRevisionIsLargerThanModRevision(t, responses)
+}
+
+func validateDuplicateHeaderRevisionMeansNoEvents(t *testing.T, responses []watchResponse) {
+	var lastHeadRevision int64 = 1
+	for _, resp := range responses {
+		if resp.Header.Revision == lastHeadRevision && len(resp.Events) != 0 {
+			t.Errorf("Got two non-empty responses about same revision")
+		}
+		lastHeadRevision = resp.Header.Revision
+	}
+}
+
+func validateResponseHeaderRevisionNeverDecreases(t *testing.T, responses []watchResponse) {
+	var lastHeadRevision int64 = 1
 	for _, resp := range responses {
 		if resp.Header.Revision < lastHeadRevision {
 			t.Errorf("Server revision should never decrease, lastHeadRevision: %d, resp.Header.Revision: %d",
 				lastHeadRevision, resp.Header.Revision)
 		}
+		lastHeadRevision = resp.Header.Revision
+	}
+}
 
+func validateGotProgressNotify(t *testing.T, responses []watchResponse, expectProgressNotify bool) {
+	var gotProgressNotify = false
+	var lastHeadRevision int64 = 1
+	for _, resp := range responses {
 		if resp.IsProgressNotify() && resp.Header.Revision == lastHeadRevision {
 			gotProgressNotify = true
 		}
-		if resp.Header.Revision == lastHeadRevision && len(resp.Events) != 0 {
-			t.Errorf("Got two non-empty responses about same revision")
-		}
+		lastHeadRevision = resp.Header.Revision
+	}
+	if gotProgressNotify != expectProgressNotify {
+		t.Errorf("Expected progress notify: %v, got: %v", expectProgressNotify, gotProgressNotify)
```

Comment:
Why do we expect progress notify in each response. 
Would be good to give more hints in the error message, or at least a comment.